### PR TITLE
PR: Move all tests to be pytest's

### LINKED ISSUE #3195: Test stand alone widgets in application
Move/copy adapt all tests or stand alone widgets to a separate tests folder using pytest-qt.

### PR DESCRIPTION
**Pull request by dalthviz**
Fixes #3195

### CODE REVIEW

**Comment by ccordoba12** [comment]
Please remove all commits related to `formlayout`. I think we don't use it anywhere in our code, and we probably should remove it (in 3.2, I think).

**Comment by ccordoba12** [comment]
@dalthviz, is this ready?

**Comment by dalthviz** [comment]
@ccordoba12 I think that there are some more tests to be moved in `utils` and `app` directories.

**Comment by ccordoba12** [comment]
@dalthviz, please remove all real tests (i.e. the ones that contain `assert` statements) in files that are not widgets.

That would be the last thing to do before merging this one. Great work, as always!!

**Comment by ccordoba12** [comment]
@dalthviz, is this one ready?

**Comment by dalthviz** [comment]
@ccordoba12 yes I think so 👍

File: spyder/utils/introspection/test/test_fallback_plugin.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+                                                        all_editable_exts,
+                                                        get_parent_until,
+                                                        python_like_mod_finder
+                                                        )
+from spyder.utils.introspection.manager import CodeInfo
+
+FALLBACK_PLUGIN_FILE = osp.join(os.path.dirname(__file__), '..',
+                                'fallback_plugin.py')
+
+def test_fallback_plugin():
```
Comment (ccordoba12) [inline spyder/utils/introspection/test/test_fallback_plugin.py]: Please break this test in several tests, instead a leaving it as a single, big one.

File: spyder/utils/introspection/test/test_fallback_plugin.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    """Test the fallback plugin."""
+    p = FallbackPlugin()
+    print(FALLBACK_PLUGIN_FILE)
```
Comment (ccordoba12) [inline spyder/utils/introspection/test/test_fallback_plugin.py]: Remove print

File: spyder/utils/introspection/test/test_module_completion.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+# Test library imports
+import pytest
+
+# Local imports
+from spyder.utils.introspection.module_completion import (module_completion, 
+                                                          get_preferred_submodules)
+from spyder.py3compat import PY3
+
+def test_codeanalysis():
```
Comment (ccordoba12) [inline spyder/utils/introspection/test/test_module_completion.py]: Change this to `test_modulecompletion`

File: spyder/utils/introspection/test/test_plugin_server.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+from spyder.utils.introspection.plugin_server import PluginServer
+
+@pytest.fixture
+def setup_plugin_server(qtbot):
+    "Setup the plugin in a separate proccess."    
+    args = sys.argv[1:]
+    if not len(args) == 2:
+        return
+    plugin = PluginServer(*args)
+    qtbot.addWidget(plugin)
```
Comment (ccordoba12) [inline spyder/utils/introspection/test/test_plugin_server.py]: `plugin_server` is not a Qt widget. Please fix this.

File: spyder/utils/introspection/test/test_utils.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+# Standard library imports
+import pickle
+
+# Test library imports
+import pytest
+
+# Local imports
+from spyder.utils.introspection.utils import CodeInfo
+
+def test_codeanalysis():
```
Comment (ccordoba12) [inline spyder/utils/introspection/test/test_utils.py]: Change this to `test_codeinfo`

File: spyder/utils/tests/test_dochelpers.py [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+# Test library imports
+import pytest
+
+# Local imports
+from spyder.utils.dochelpers import getargtxt, getdoc, getobj, isdefined
+from spyder.py3compat import PY2
+
+def test_dochelpers():
+    """Test dochelpers."""
+    class Test(object):
```
Comment (ccordoba12) [inline spyder/utils/tests/test_dochelpers.py]: Please move this class out of the test definition

File: spyder/utils/tests/test_dochelpers.py [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+                                                   'order.\n\nA custom '
+                                                   'key function can be '
+                                                   'supplied to customize the '
+                                                   'sort order, and '
+                                                   'the\nreverse flag can be '
+                                                   'set to request the result '
+                                                   'in descending order.',
+                                                   'name': 'sorted'}          
+        assert getargtxt(sorted) == None
+    assert isdefined('numpy.take', force_import=True) == True
```
Comment (ccordoba12) [inline spyder/utils/tests/test_dochelpers.py]: There's no need to `assert ... == True`. You can change this simply to `assert ...`

File: spyder/utils/tests/test_dochelpers.py [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+    assert isdefined('__import__') == True
```
Comment (ccordoba12) [inline spyder/utils/tests/test_dochelpers.py]: Remove `True`

File: spyder/utils/tests/test_dochelpers.py [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+    assert isdefined('.keys', force_import=True) == False
```
Comment (ccordoba12) [inline spyder/utils/tests/test_dochelpers.py]: `assert not ...` and remove `False`

**Comment by ccordoba12** [comment]
The failure in CircleCI is caused by me. Don't worry about it :-)

File: spyder/utils/introspection/test/test_modulecompletion.py [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+
+# Test library imports
+import pytest
+
+# Local imports
+from spyder.utils.introspection.module_completion import (module_completion, 
+                                                          get_preferred_submodules)
+from spyder.py3compat import PY3
+
+def test_codeanalysis():
```
Comment (ccordoba12) [inline spyder/utils/introspection/test/test_modulecompletion.py]: `test_codeanalysis` -> `test_module_completion`

**Comment by ccordoba12** [comment]
Please merge with `3.1.x` and solve the little error I mentioned above.

**Pull request merged**